### PR TITLE
Add ZH/LT-01 climate component with IR receiver option

### LINKED ISSUE #1050: IR Climate support ZH/JT01 ZH/JT03 ( Chigo Komeco Lloyd Blaupunkt EPiK )
**Describe the problem you have/What new integration you would like**

Both way support for ZH/JT01 /JT03 remotes - they are used by many brands -> protocol codes https://github.com/arthurkrupa/ac-ir-mqtt-zhjt03/blob/master/include/codes.h 

**Please describe your use case for this integration and alternatives you've tried:**

Works with https://github.com/arthurkrupa/ac-ir-mqtt-zhjt03

**Additional context**

Extension of supported remotes

### PR DESCRIPTION
**Pull request by cfeenstra1024**
# What does this implement/fix?

Added climate integration for AC units using the 'ZH/LT-01' remote control. The integration includes support for an IR receiver.

The ZH/LT-01 is a remote control that is used with many locally branded split airconditioners: Eurom, Chigo, Tristar, Tecnomaster, Elgin, Geant, Tekno, Topair, Proma, Sumikura, JBS, Turbo Air, Nakatomy, Celestial Air, Ager, Blueway, Airlux, etc

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes [feature request 1050](https://github.com/esphome/feature-requests/issues/1050)

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2204

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
remote_transmitter:
  pin: GPIO5

remote_receiver:
  id: ac_recv
  pin:
    number: GPIO4
    inverted: true
    mode:
      input: true
      pullup: true
  tolerance: 45%
  filter: 250us

climate:
  - platform: zhlt01
    name: "DEV 8266 AC2"
    supports_heat: false
    supports_cool: true
    receiver_id: ac_recv
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by cfeenstra1024** [comment]
Recreated PR after the [original one](https://github.com/esphome/esphome/pull/3668) got closed after half a year. I would however like this to be merged if possible.

**Comment by cfeenstra1024** [comment]
https://github.com/esphome/esphome/pull/4333#issuecomment-1518447539

Would be nice if this can be reviewed and merged

**Comment by cfeenstra1024** [comment]
Just a comment to keep the PR alive.
Would be nice if this can be merged ;)

**Review by jesserockz — DISMISSED** [review]

**Review by jesserockz — DISMISSED** [review]

File: esphome/components/zhlt01/zhlt01.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      break;
+    case climate::CLIMATE_MODE_FAN_ONLY:
+      ir_message[9] = AC1_MODE_FAN;
+      break;
+    default:
+      break;
+  }
+
+  // -- Temperature
+  ir_message[9] |= (uint8_t)(this->target_temperature - 16.0f);
```
Comment (jesserockz) [inline esphome/components/zhlt01/zhlt01.cpp]:
```suggestion
  ir_message[9] |= (uint8_t) (this->target_temperature - 16.0f);
```

File: esphome/components/zhlt01/zhlt01.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  // Validate footer
+  if (!data.expect_mark(AC1_BIT_MARK)) {
+    ESP_LOGV(TAG, "Footer fail");
+    return false;
+  }
+
+  // Validate checksum
+  for (int i = 0; i < 12; i += 2) {
+    if (ir_message[i] != (uint8_t)(~ir_message[i + 1])) {
+      ESP_LOGV(TAG, "Byte %d checksum incorrect (%02X != %02X)", i, ir_message[i], (uint8_t)(~ir_message[i + 1]));
```
Comment (jesserockz) [inline esphome/components/zhlt01/zhlt01.cpp]:
```suggestion
    if (ir_message[i] != (uint8_t) (~ir_message[i + 1])) {
      ESP_LOGV(TAG, "Byte %d checksum incorrect (%02X != %02X)", i, ir_message[i], (uint8_t) (~ir_message[i + 1]));
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**